pkg/registry/apis/iam/legacy/user.go

```diff
@@ -287,3 +291,357 @@ func (s *legacySQLStore) ListUserTeams(ctx context.Context, ns claims.NamespaceI
 
 	return res, err
 }
+
+type CreateUserCommand struct {
+	UID           string
+	Email         string
+	Login         string
+	Name          string
+	OrgID         int64
+	IsAdmin       bool
+	IsDisabled    bool
+	EmailVerified bool
+	IsProvisioned bool
+	Salt          string
+	Rands         string
+	Created       time.Time
+	Updated       time.Time
+	LastSeenAt    time.Time
+	Role          string
+}
+
+type CreateUserResult struct {
+	User user.User
+}
+
+type CreateOrgUserCommand struct {
+	OrgID   int64
+	UserID  int64
+	Role    string
+	Created time.Time
+	Updated time.Time
+}
+
+type DeleteUserCommand struct {
+	UID string
+}
+
+type DeleteUserQuery struct {
+	OrgID int64
+	UID   string
+}
+
+type DeleteUserResult struct {
+	Success bool
+}
+
+var sqlCreateUserTemplate = mustTemplate("create_user.sql")
+var sqlCreateOrgUserTemplate = mustTemplate("create_org_user.sql")
+var sqlDeleteUserTemplate = mustTemplate("delete_user.sql")
+var sqlDeleteOrgUserTemplate = mustTemplate("delete_org_user.sql")
+
+func newCreateUser(sql *legacysql.LegacyDatabaseHelper, cmd *CreateUserCommand) createUserQuery {
+	return createUserQuery{
+		SQLTemplate:  sqltemplate.New(sql.DialectForDriver()),
+		UserTable:    sql.Table("user"),
+		OrgUserTable: sql.Table("org_user"),
+		Command:      cmd,
+	}
+}
+
+type createUserQuery struct {
+	sqltemplate.SQLTemplate
+	UserTable    string
+	OrgUserTable string
+	Command      *CreateUserCommand
+}
+
+func (r createUserQuery) Validate() error {
+	if r.Command.Login == "" && r.Command.Email == "" {
```

Comment:
I think you'll want these kinds of validations as a validation webhook instead, so regardless of if it is stored to unified storage, or to legacy storage, these checks will happen

for example, define which methods you want to do validation on: https://github.com/grafana/grafana/blob/main/apps/playlist/pkg/apis/playlist_manifest.go#L29
And then you can do the actual functions here: https://github.com/grafana/grafana/blob/main/apps/playlist/pkg/app/app.go#L62